scripts/multi_component_hydrogen_bond_propensity/multi_component_hydrogen_bond_propensity_report.py

```diff
@@ -190,13 +190,16 @@ def format_scores(scores, das, d_type):
     return formatted_scores
 
 
-def get_mc_scores(propensities, identifier):
+def get_mc_scores(propensities, identifier, ignore_intra:bool):
     # Calculates the multi-component scores from the individual HBP calculation
     AA_propensities = []
     BB_propensities = []
     AB_propensities = []
     BA_propensities = []
 
+    if ignore_intra is True:
+        propensities = [p for p in propensities if p.is_intermolecular]
+
     for p in propensities:
         t = "%s_d" % p.donor_label.split(" ")[0], "%s_a" % p.acceptor_label.split(" ")[0]
         if '_A_' in t[0] and '_A_' in t[1]:
```

Comment:
A stylistic suggestion: It might be worth just moving this into the for loop before, as you are already going through each thing a time. 
```python
for p in propensities:
    if ignore_intra and not p.is_intermolecular:
        continue
```

Reply:
Wouldn't this need to be 
`if ignore_intra and p.is_intermolecular:` ?

that if statement currently returns False for an intermolecular H-bond

Reply:
An alternative would be to do this around line 99 where you calculate and return the propensities